tests/unit/test_utils.py

```diff
@@ -627,6 +627,28 @@ def test_call_subprocess_closes_stdin():
         call_subprocess([sys.executable, '-c', 'input()'])
 
 
+@pytest.mark.parametrize('args, expected', [
+    # Test without subdir.
+    (('git+https://example.com/pkg', 'dev', 'myproj'),
+     'git+https://example.com/pkg@dev#egg=myproj'),
+    # Test with subdir.
+    (('git+https://example.com/pkg', 'dev', 'myproj', 'sub/dir'),
+     'git+https://example.com/pkg@dev#egg=myproj&subdirectory=sub/dir'),
+    # Test with None subdir.
+    (('git+https://example.com/pkg', 'dev', 'myproj', None),
+     'git+https://example.com/pkg@dev#egg=myproj'),
+])
+def test_make_vcs_requirement_url(args, expected):
+    if len(args) == 3:
```

Comment:
Perhaps this test can probably be simplified by simply using *args here.

This works as is though, I'm just wondering if there's a cleaner way to write this.

Reply:
Can you suggest / think of something cleaner? When I considered this, I thought there was value in separating the input arguments from the expected value using parentheses, as opposed to having everything as a single list in the test parameterization.

Reply:
Oh, I see. Let me propose something different.

Reply:
Yeah, this is what I was thinking. It's much nicer. :)

Reply:
I was wanting to pass `subdir` as a keyword argument, but it's not critical I guess.